shell/platform/embedder/fixtures/main.dart

```diff
@@ -178,3 +179,78 @@ void null_platform_messages() {
   };
   signalNativeTest();
 }
+
+Picture CreateSimplePicture() {
+  Paint blackPaint = Paint();
+  PictureRecorder baseRecorder = PictureRecorder();
+  Canvas canvas = Canvas(baseRecorder);
+  canvas.drawRect(Rect.fromLTRB(0.0, 0.0, 1000.0, 1000.0), blackPaint);
+  return baseRecorder.endRecording();
+}
+
+@pragma('vm:entry-point')
+void can_composite_platform_views() {
+  window.onBeginFrame = (Duration duration) {
+    SceneBuilder builder = SceneBuilder();
+    builder.addPicture(Offset(1.0, 1.0), CreateSimplePicture());
+    builder.pushOffset(1.0, 2.0);
+    builder.addPlatformView(42, width: 123.0, height: 456.0);
+    builder.addPicture(Offset(1.0, 1.0), CreateSimplePicture());
+    builder.pop(); // offset
+    signalNativeTest(); // Signal 2
+    window.render(builder.build());
+  };
+  signalNativeTest(); // Signal 1
+  window.scheduleFrame();
+}
+
+Picture CreateColoredBox(Color color, Size size) {
+  Paint paint = Paint();
+  paint.color = color;
+  PictureRecorder baseRecorder = PictureRecorder();
+  Canvas canvas = Canvas(baseRecorder);
+  canvas.drawRect(Rect.fromLTRB(0.0, 0.0, size.width, size.height), paint);
+  return baseRecorder.endRecording();
+}
+
+@pragma('vm:entry-point')
+void can_composite_platform_views_with_known_scene() {
+  window.onBeginFrame = (Duration duration) {
+    Color red = Color.fromARGB(127, 255, 0, 0);
+    Color blue = Color.fromARGB(127, 0, 0, 255);
+    Color gray = Color.fromARGB(127, 127, 127, 127);
+
+    Size size = Size(100.0, 100.0);
```

Comment:
nit: I'd make width != height to catch a possible faulty swap of these parameters.

Reply:
Roger. Update fixtures and expectations.